dotCMS/src/main/java/com/dotmarketing/portlets/structure/factories/RelationshipCache.java

```diff
@@ -50,5 +51,38 @@ public void removeRelationshipsByType(ContentTypeIf type) {
 		// TODO Auto-generated method stub
 		return null;
 	}
-	
+
+    /**
+     * Gets related content for a given contentlet
+     * @param contentletIdentifier
+     * @return
+     * @throws DotCacheException
+     */
+    public abstract Map<String, List<String>> getRelatedContentMap(String contentletIdentifier)
+            throws DotCacheException;
+
+    /**
+     * Adds related content to the cache for a given contentlet
+     * @param contentletIdentifier
+     * @param relatedContent
+     */
+    public abstract void putRelatedContentMap(String contentletIdentifier,
+            Map<String, List<String>> relatedContent);
+
+    /**
+     * Invalidates relationship cache for a given contentlet
+     * @param contentletIdentifier
+     */
+    public abstract void removeRelatedContentMap(String contentletIdentifier);
+
+    /**
+     * Removes related content from cache given a contentlet identifier and the velocity var name of
+     * the relationship field
+     * @param contentletIdentifier
+     * @param relationshipFieldVar
+     * @throws DotCacheException
+     */
+    public abstract void removeRelatedContentFromMap(String contentletIdentifier,
+            String relationshipFieldVar)
+            throws DotCacheException;

```

Comment:
do we need the `throws DotCacheException;` in the method signature?

Reply:
we don't @wezell , I just kept the convention used by other methods

Reply:
I mean, I need to handle the exception. I could catch it or throw it. I preferred to throw it to keep the convention